src/utils.rs

```diff
@@ -200,15 +218,25 @@ pub async fn estimate_gas_and_price(
     }
 
     while Instant::now() < deadline {
-        let estimated_gas = http_rpc_client.estimate_gas(txn, None).await;
+        // Estimate the gas required for the TransactionRequest from the rpc, retry otherwise
+        let estimated_gas = http_rpc_client
+            .estimate_gas(&TypedTransaction::Legacy(txn.to_owned()), None)
```

Comment:
Does the old method involving abigen (FunctionCall struct) produce the legacy transaction? How does it decide?

Reply:
Juts checked that the `abigen!` produced `Eip1559` transaction types. `Legacy` type is still okay to use while estimating gas since the extra fields that are present in `Eip1559` type i.e., `max_priority_fee_per_gas` and `max_fee_per_gas` are not taken into account there. For sending the transactions, will make the change to make the transaction type `Eip1559` so that the signer rpc client will fill those extra fields and proceed accordingly.